tests/cases/compiler/divergentAccessorsTypes8.ts

```diff
@@ -0,0 +1,153 @@
+// @strict: true
+// @lib: esnext, dom
+// @noEmit: true
+
+export {}
+
+interface Serializer {
+  set value(v: string | number | boolean);
+  get value(): string;
+}
+declare let box: Serializer;
+const v = box['value']
+box['value'] = true;
+box['value'] = 42;
+box['value'] = "hello";
+
+interface Element {
+    get style(): CSSStyleDeclaration;
+    set style(cssText: string);
+}
+
+declare const element: Element;
+element['style'] = "color: red";
+element['style'] = element.style;
+
+class One {
+  get prop1(): string {
+    return "";
+  }
+  set prop1(s: string | number) {}
+
+  get prop2(): string {
+    return "";
+  }
+  set prop2(s: string | number) {}
+
+  prop3: number = 42;
+
+  get prop4(): string {
+    return "";
+  }
+  set prop4(s: string | number) {}
+}
+
+class Two {
+  get prop1(): string {
+    return "";
+  }
+  set prop1(s: string | number) {}
+
+  get prop2(): string {
+    return "";
+  }
+  set prop2(s: string) {}
+
+  get prop3(): string {
+    return "";
+  }
+  set prop3(s: string | boolean) {}
+
+  get prop4(): string {
+    return "";
+  }
+  set prop4(s: string | boolean) {}
+}
+
+declare const u1: One | Two;
+
+u1['prop1'] = 42;
+u1['prop1'] = "hello";
+
+u1['prop2'] = 42;
```

Comment:
I think that this one should error. See the playground [here](https://www.typescriptlang.org/play?ts=5.3.0-dev.20230831#code/MYGwhgzhAEDyB2BTaBvAUNaBzRAXaADgE4D2BATABQCUAXNBLkQJbxaoabRF4CuR8aACIhAbk4BfThDyFSFShHqMWbaAB9o8XgFsARoiLVUUqWlCQYAFQDuJDphz5iZKnQZNW7dF258BwmKS0rIuCkoeqljGKKZoaAAmiBY80MAk8IzQGYj0CIjiOQB0YeTQALzQACzk4onJ4Knpmfi4dvS2JOJtJCXyZZU1dUkpyM1ZvPDMGXlIGtCd4pPT8H2uFdW1aEA)

I re-verified all of the tests here and everything here behaves in the same way as it behaves for regular property access expressions. So I think that this issue is a separate one and I will try to prepare a follow up PR to fix it.